pkg/httputil/handler.go

```diff
@@ -13,3 +15,16 @@ func WrapHandlerFunc(handlerFn func(req *http.Request) error) http.HandlerFunc {
 		}
 	})
 }
+
+// HandlerNotImplementedIfSettingDisabled takes a handler func and returns a handler func that will error out if
+// specified boolean is disabled, otherwise it will return the given handler func
+func HandlerNotImplementedIfSettingDisabled(enabled bool, fn http.Handler) http.Handler {
```

Comment:
This function seems oddly specific for a utility package.
1. Why is it relevant that the `bool` parameter corresponds to a _setting_?
2. What if one wanted to customize the error message?

I think a more elegant way to solve this might be by introducing a general-purpose ITE (if-then-else)/ternary operator function, e.g., in `pkg/utils`, like:
```go
func ITE[T any](cond bool, a, b T) T {
  if cond {
    return a
  }
  return b
}
```
and then also have something like a `func NotImplementedHandler(msg string) http.Handler`. You can then invoke this as, e.g.,
```go
utils.ITE[http.Handler](env.DisableKernelPackageUpload.BooleanSetting(), httputil.NotImplementedHandler("..."), http.HandlerFunc(s.handleProbeUpload))
```

Reply:
I totally forgot we support generics now. Love this idea so I'll switch to this